app/src/main/java/com/keylesspalace/tusky/usecase/TrendingCases.kt

```diff
@@ -0,0 +1,48 @@
+/* Copyright 2023 Tusky Contributors
+ *
+ * This file is a part of Tusky.
+ *
+ * This program is free software; you can redistribute it and/or modify it under the terms of the
+ * GNU General Public License as published by the Free Software Foundation; either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * Tusky is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even
+ * the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General
+ * Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with Tusky; if not,
+ * see <http://www.gnu.org/licenses>. */
+
+package com.keylesspalace.tusky.usecase
+
+import android.util.Log
+import com.keylesspalace.tusky.entity.TrendingTag
+import com.keylesspalace.tusky.network.MastodonApi
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import javax.inject.Inject
+
+/**
+ * Created by @knossos@fosstodon.org on 2023-01-07.
+ */
+class TrendingCases @Inject constructor(
+    private val mastodonApi: MastodonApi,
+) {
+    suspend fun trendingTags(): List<TrendingTag> {
```

Comment:
Why do you have this layer of indrection? The only caller is in TrendingViewModel, so you could `MastodonApi` there, and call the API's `trendingTags()` function directly, in the viewmodel's scope.

Am I missing something subtle?

Reply:
Having the cases class allows us to replace the functionality for testing purposes. API calls don't belong in the ViewModel ideally.

I would usually implement a Repository (in a similar style to InstanceInfoRepository), but since I was copying the Timeline to a degree, I ended up using the cases class as a style guide (see: TimelineCases).

We can implement the other trending APIs in the same class.

What do you think?

Reply:
Sure, I'll remove it :+1: